src/Spinners/__tests__/LoadSpinner.test.tsx

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import { render } from 'test-utils';
+import LoadSpinner from '../LoadSpinner';
+
+describe('Testing Data Loading Spinner', function () {
+  let getByTestId;
+  let getByText;
+  beforeEach(function () {
+    ({ getByTestId, getByText } = render(
+      <LoadSpinner>
+        <div data-testid="test-spinner"> Data Loading  </div>
+      </LoadSpinner>
+    ));
+  });
+
+  it('renders', function () {
+    getByTestId('test-spinner');
+  });
+
+  it('renders value', function () {
+    getByText('Data Loading');
+  });
```

Comment:
Both of these are really testing the same thing, so I'd suggest only using the `getByText` one since that's close to how a user would interact with this component.

That being said, lately I've been feeling that these kinds of tests ("renders what we put into it") aren't _actually_ all that useful. It might make sense to use the `container` property returned from render to do something like `container.querySelector('.fa-spinner.fa-spin')` and make sure that the icon is actually there. I don't know, I go back and forth over how much it's actually worth testing UI components like this...

Reply:
I was thinking about the same exact thing, how much the test is useful and how deep I should go with the testing ,  I event started the container query selector and remove it. I guess we can add more tests as needed (if needed) .